src/component/Cursor.tsx

```diff
@@ -0,0 +1,95 @@
+import React, { ReactElement, cloneElement, createElement, isValidElement } from 'react';
+import { ChartCoordinate, ChartOffset, LayoutType, TooltipEventType } from '../util/types';
+import { Curve } from '../shape/Curve';
+import { Cross } from '../shape/Cross';
+import { getCursorRectangle } from '../util/cursor/getCursorRectangle';
+import { Rectangle } from '../shape/Rectangle';
+import { getRadialCursorPoints } from '../util/cursor/getRadialCursorPoints';
+import { Sector } from '../shape/Sector';
+import { getCursorPoints } from '../util/cursor/getCursorPoints';
+import { filterProps } from '../util/ReactUtils';
+
+export type CursorProps = {
+  activeCoordinate: ChartCoordinate;
+  activePayload: any[];
+  activeTooltipIndex: number;
+  chartName: string;
+  element: ReactElement;
+  isActive: boolean;
+  layout: LayoutType;
+  offset: ChartOffset;
+  tooltipAxisBandSize: number;
+  tooltipEventType: TooltipEventType;
+};
+
+/*
+ * Cursor is the background, or a highlight,
+ * that shows when user mouses over or activates
+ * an area.
+ *
+ * It usually shows together with a tooltip
+ * to emphasise which part of the chart does the tooltip refer to.
+ */
+export function Cursor(props: CursorProps) {
+  const {
+    element,
+    tooltipEventType,
+    isActive,
+    activeCoordinate,
+    activePayload,
+    offset,
+    activeTooltipIndex,
+    tooltipAxisBandSize,
+    layout,
+    chartName,
+  } = props;
+  if (
+    !element ||
+    !element.props.cursor ||
+    !isActive ||
+    !activeCoordinate ||
+    (chartName !== 'ScatterChart' && tooltipEventType !== 'axis')
```

Comment:
It'll be important later on to not forget logic such as this line (and below) exists. Not sure if we should plan for some mechanism to separate the differences (I know this PR is just a step so all good)